rllib/algorithms/marwil/marwil.py

```diff
@@ -175,6 +221,33 @@ def evaluation(
 
         return self
 
+    @override(AlgorithmConfig)
+    def offline_data(self, **kwargs) -> "MARWILConfig":
```

Comment:
We should not do this here.

The config API should NOT rely on users calling this method, but should set these defaults in the algo itself (where we are actually building the PreLearner). See `config.env_runner_class` for comparison (we only decide inside the EnvRunnerGroup, right before we have to construct the n EnvRunners, which class we should use).

Solution: Move the logic below into the main Algorithm code (I guess bc.py or marwil.py).

Reply:
This would mean that the `OfflineData`  class or `Algorithm` (during setup) need to know about `MARWIL` and `CQL` to call their prelearner classes. This would mix things together which might be better to be separated. 

I would like to avoid to override the Algorithm's setup method. Where would you include this?